scripts/Diffraction/isis_powder/polaris.py

```diff
@@ -75,13 +75,13 @@ def ensure_per_detector_and_vanadium_output_are_in_sync(self, vanadium_d, per_de
             )
 
     def create_total_scattering_pdf(self, **kwargs):
-        if "pdf_type" not in kwargs or kwargs["pdf_type"] not in ["G(r)", "g(r)", "RDF(r)", "G_k(r)"]:
-            kwargs["pdf_type"] = "G(r)"
+        self._inst_settings.update_attributes(kwargs=kwargs)
+        if self._inst_settings.pdf_type is None or self._inst_settings.pdf_type not in ["G(r)", "g(r)", "RDF(r)", "G_k(r)"]:
```

Comment:
I think this is working properly now. Before when I ran the test script I didn't need to specify the `pdf_type` in the function `create_total_scattering_pdf` even though it wasn't in the config file (it defaulted to `G(r)`). Now it throws an error.

Reply:
Do you get an error or a warning? If no pdf type is specified it should default it to G(r) and provide a warning

Reply:
Ah sorry I got an error:
```
AttributeError: The parameter with name: 'pdf_type' is required but was not set or passed.

 File "C:/Users/xhg73778/OneDrive - Science and Technology Facilities Council/test_polaris.py", line 17, in <module>
 pol.create_total_scattering_pdf(run_number="98533", merge_banks=False)
 File "C:\mantid-conda\mantid\scripts\Diffraction\isis_powder\polaris.py", line 79, in create_total_scattering_pdf
 if self._inst_settings.pdf_type is None or self._inst_settings.pdf_type not in ["G(r)", "g(r)", "RDF(r)", "G_k(r)"]:
 File "C:\mantid-conda\mantid\scripts\Diffraction\isis_powder\routines\instrument_settings.py", line 62, in __getattr__
 self._raise_user_param_missing_error(known_param)
 File "C:\mantid-conda\mantid\scripts\Diffraction\isis_powder\routines\instrument_settings.py", line 124, in _raise_user_param_missing_error
 raise AttributeError(err_text)
```

Reply:
OK. I think the offending line needs a `hasattr(self._inst_settings, "pdf_type")` added to the if statement.

The problem I'd tried to fix was that if the pdf type was specified in the confi file or in the line of your script where the Polaris object is created, it was ignored and the default G(r) still kicked in.